packages/tailwindcss/src/ast.ts

```diff
@@ -202,7 +202,7 @@ export function walkDepth(
   }
 }
 
-export function toCss(ast: AstNode[]) {
+export function toCss(ast: AstNode[], { printUtilitiesNode = false } = {}) {
```

Comment:
Still feel conflicted about this. Like we talked, this should ultimately not be a concern of the printer (the printer should be agnostic to Tailwind, really). 

Maybe one other idea is that we allow the call sites where we want the `@tailwind utilities` directives to be removed to _mutate_ the `at-rule` with something like `params = ''` or so and then use that as hint to not print it? 

So basically right before [this line](https://github.com/tailwindlabs/tailwindcss/blob/next/packages/tailwindcss/src/index.ts#L509) we do something like this:

```ts
if (utilitiesNode) { 
  // Avoid printing the utility node
  utilitiesNode.name = ''
}
```

Then in the `toCSS` we skip at-rules with `name === ''`. Since the `at-rule` always contains the prefixed `@` this is really not valid otherwise anyways. WDYT?

Reply:
Yeah I still don't like this either. I kinda still want a "contents" node of sorts that only prints its children and we'd "switch" the utility node to it in `compile()`. I wonder if we could use the existing context node for that because it's basically that. Thoughts?

Reply:
There are other nodes being printed here that are Tailwind CSS specific. E.g.: declarations like `--tw-sort`. We could add another pass that gets rid of them.

I think since we only care about the utilitiesNode reference that we fill up with the actual utilities, I think changing it to a context node instead might be a nicer way of handling as @thecrypticace mentioned.

I also don't like the mutation of `utilitiesNode.name` or using the `printUtilitiesNode` flag. Both feel like a bandaid solution.